lte/gateway/c/oai/tasks/sgw/sgw_handlers.c

```diff
@@ -1239,101 +1240,28 @@ static void sgw_release_all_enb_related_information(
       sgw_cm_get_spgw_context(release_access_bearers_req_pP->teid);
   if (ctx_p) {
-    release_access_bearers_resp_p->cause.cause_value = REQUEST_ACCEPTED;
-    release_access_bearers_resp_p->teid =
-        ctx_p->sgw_eps_bearer_context_information.mme_teid_S11;
-    release_access_bearers_resp_p->trxn = release_access_bearers_req_pP->trxn;
-    //#pragma message  "TODO Here the release
-    //(sgw_handle_release_access_bearers_request)"
-    /*
-     * Release the tunnels so that in idle state, DL packets are not sent
-     * towards eNB.
-     * These tunnels will be added again when UE moves back to connected mode.
-     */
-    // TODO iterator
-    for (int ebx = 0; ebx < BEARERS_PER_UE; ebx++) {
-      sgw_eps_bearer_ctxt_t* eps_bearer_ctxt =
-          ctx_p->sgw_eps_bearer_context_information.pdn_connection
-              .sgw_eps_bearers_array[ebx];
-      if (eps_bearer_ctxt) {
-        struct in_addr enb = {.s_addr = 0};
-        enb.s_addr =
-            eps_bearer_ctxt->enb_ip_address_S1u.address.ipv4_address.s_addr;
-
-        struct in6_addr* ue_ipv6 = NULL;
-        if ((eps_bearer_ctxt->paa.pdn_type == IPv6) ||
-            (eps_bearer_ctxt->paa.pdn_type == IPv4_AND_v6)) {
-          ue_ipv6 = &eps_bearer_ctxt->paa.ipv6_address;
-        }
-
-        rv = gtp_tunnel_ops->del_tunnel(
-            enb, eps_bearer_ctxt->paa.ipv4_address, ue_ipv6,
-            eps_bearer_ctxt->s_gw_teid_S1u_S12_S4_up,
-            eps_bearer_ctxt->enb_teid_S1u, NULL);
-        if (rv < 0) {
-          OAILOG_ERROR_UE(
-              LOG_SPGW_APP, imsi64,
-              "ERROR in deleting TUNNEL " TEID_FMT " (eNB) <-> (SGW) " TEID_FMT
-              "\n",
-              eps_bearer_ctxt->enb_teid_S1u,
-              eps_bearer_ctxt->s_gw_teid_S1u_S12_S4_up);
-        }
-        // Paging is performed without packet buffering
-        rv = gtp_tunnel_ops->add_paging_rule(eps_bearer_ctxt->paa.ipv4_address);
-        // Convert to string for logging
-        char* ip_str = inet_ntoa(eps_bearer_ctxt->paa.ipv4_address);
-        if (rv < 0) {
-          OAILOG_ERROR(
-              LOG_SPGW_APP, "ERROR in setting paging rule for IP Addr: %s\n",
-              ip_str);
-        } else {
-          OAILOG_DEBUG(
-              LOG_SPGW_APP, "Set the paging rule for IP Addr: %s\n", ip_str);
-        }
-
-        sgw_release_all_enb_related_information(eps_bearer_ctxt);
-      }
-    }
-
-    rv = send_msg_to_task(&spgw_app_task_zmq_ctx, TASK_MME, message_p);
-
-    OAILOG_DEBUG_UE(
-        LOG_SPGW_APP, imsi64, "Release Access Bearer Response sent to MME\n");
-    OAILOG_FUNC_RETURN(LOG_SPGW_APP, rv);
+    sgw_send_release_access_bearer_response(
+        LOG_SPGW_APP, imsi64, REQUEST_ACCEPTED, release_access_bearers_req_pP,
+        ctx_p->sgw_eps_bearer_context_information.mme_teid_S11);
+    sgw_process_release_access_bearer_request(
+        LOG_SPGW_APP, imsi64, &(ctx_p->sgw_eps_bearer_context_information));
```

Comment:
Should not we sending request rejected if tunnels are not deleted successfully?

Reply:
Can you add at least a TO DO comment? This does not change the existing logic, so we can still land the PR...